src/pyhf/infer/calculators.py

```diff
@@ -806,33 +810,31 @@ def expected_pvalues(self, sig_plus_bkg_distribution, bkg_only_distribution):
               The distribution for the background-only hypothesis.
 
         Returns:
-            Tuple (:obj:`float`): The :math:`p`-values for the test statistic
+            Tuple (:obj:`tensor`): The :math:`p`-values for the test statistic
             corresponding to the :math:`\mathrm{CL}_{s+b}`,
             :math:`\mathrm{CL}_{b}`, and :math:`\mathrm{CL}_{s}`.
         """
         tb, _ = get_backend()
-        pvalues = tb.astensor(
-            [
-                self.pvalues(
-                    tb.astensor(test_stat),
-                    sig_plus_bkg_distribution,
-                    bkg_only_distribution,
-                )
-                for test_stat in bkg_only_distribution.samples
-            ]
-        )
+        pvalues = [
+            self.pvalues(
+                test_stat,
+                sig_plus_bkg_distribution,
+                bkg_only_distribution,
+            )
+            for test_stat in bkg_only_distribution.samples
+        ]
         # TODO: Add percentile to tensorlib
         # c.f. Issue #815, PR #817
         import numpy as np
 
         # percentiles for -2, -1, 0, 1, 2 standard deviations of the Normal distribution
         normal_percentiles = [2.27501319, 15.86552539, 50.0, 84.13447461, 97.72498681]
         pvalues_exp_band = np.percentile(
-            tb.tolist(pvalues),
+            pvalues,
```

Comment:
wouldn't this break for the other backends if `np.percentile` can't run on the tensorflow/jax/pytorch arrays correctly? That's why the `tolist` exists -- as we don't have the global tensorlib.percentile yet.

Reply:
@kratsg I thought the same, but `pvalues` here is already a list, so calling `tb.tolist` on the list just returns the list again

```python
>>> import pyhf
>>> pyhf.set_backend("pytorch")
>>> x = [pyhf.tensorlib.astensor(x) for x in range(0, 10)]
>>> x
[tensor(0.), tensor(1.), tensor(2.), tensor(3.), tensor(4.), tensor(5.), tensor(6.), tensor(7.), tensor(8.), tensor(9.)]
>>> y = pyhf.tensorlib.tolist(x)
>>> y
[tensor(0.), tensor(1.), tensor(2.), tensor(3.), tensor(4.), tensor(5.), tensor(6.), tensor(7.), tensor(8.), tensor(9.)]
```

so this seems to be the behavior we already had

```python
import pyhf
import numpy.random as random

random.seed(0)

for backend in ["numpy", "pytorch", "jax", "tensorflow"]:
    print(f"\nbackend: {backend}")
    pyhf.set_backend(backend)
    model = pyhf.simplemodels.hepdata_like(
        signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0]
    )
    observations = [51, 48]
    data = observations + model.config.auxdata
    mu_test = 1.0
    toy_calculator = pyhf.infer.calculators.ToyCalculator(
        data, model, ntoys=100, track_progress=False
    )
    sig_plus_bkg_dist, bkg_dist = toy_calculator.distributions(mu_test)
    CLsb_exp_band, CLb_exp_band, CLs_exp_band = toy_calculator.expected_pvalues(
        sig_plus_bkg_dist, bkg_dist
    )

    print(f"CLsb_exp_band: {CLsb_exp_band}")
    print(f"CLb_exp_band: {CLb_exp_band}")
    print(f"CLs_exp_band: {CLs_exp_band}")
```
gives

```
backend: numpy
CLsb_exp_band: [array(0.), array(0.), array(0.03), array(0.2387939), array(1.)]
CLb_exp_band: [array(0.03252263), array(0.1670687), array(0.505), array(0.8429313), array(1.)]
CLs_exp_band: [array(0.), array(0.), array(0.06186224), array(0.28450033), array(1.)]

backend: pytorch
CLsb_exp_band: [tensor(0.), tensor(0.), tensor(0.0300), tensor(0.2629), tensor(0.8562)]
CLb_exp_band: [tensor(0.0325), tensor(0.1671), tensor(0.5050), tensor(0.8429), tensor(0.9924)]
CLs_exp_band: [tensor(0.), tensor(0.), tensor(0.0690), tensor(0.3119), tensor(0.8596)]

backend: jax
WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
CLsb_exp_band: [DeviceArray(0., dtype=float64), DeviceArray(0.01, dtype=float64), DeviceArray(0.02, dtype=float64), DeviceArray(0.23, dtype=float64), DeviceArray(1., dtype=float64)]
CLb_exp_band: [DeviceArray(0.03252263, dtype=float64), DeviceArray(0.1670687, dtype=float64), DeviceArray(0.505, dtype=float64), DeviceArray(0.8429313, dtype=float64), DeviceArray(1., dtype=float64)]
CLs_exp_band: [DeviceArray(0., dtype=float64), DeviceArray(0.03004177, dtype=float64), DeviceArray(0.04919355, dtype=float64), DeviceArray(0.26836416, dtype=float64), DeviceArray(1., dtype=float64)]

backend: tensorflow
CLsb_exp_band: [<tf.Tensor: shape=(), dtype=float32, numpy=0.0>, <tf.Tensor: shape=(), dtype=float32, numpy=0.0>, <tf.Tensor: shape=(), dtype=float32, numpy=0.04>, <tf.Tensor: shape=(), dtype=float32, numpy=0.14>, <tf.Tensor: shape=(), dtype=float32, numpy=0.40981895>]
CLb_exp_band: [<tf.Tensor: shape=(), dtype=float32, numpy=0.03252263>, <tf.Tensor: shape=(), dtype=float32, numpy=0.1670687>, <tf.Tensor: shape=(), dtype=float32, numpy=0.505>, <tf.Tensor: shape=(), dtype=float32, numpy=0.8429313>, <tf.Tensor: shape=(), dtype=float32, numpy=0.9774774>]
CLs_exp_band: [<tf.Tensor: shape=(), dtype=float32, numpy=0.0>, <tf.Tensor: shape=(), dtype=float32, numpy=0.0>, <tf.Tensor: shape=(), dtype=float32, numpy=0.06957048>, <tf.Tensor: shape=(), dtype=float32, numpy=0.16927768>, <tf.Tensor: shape=(), dtype=float32, numpy=0.4191114>]
```

Reply:
@kratsg I could also throw in

```
pvalues = [[tb.tolist(pvalue) for pvalue in band] for band in pvalues]
```

but that just gives the same results we already have and takes about the same amount of time, or slower.